Crash in eth1deposits handler

This happened on chaind 0.6.10.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb3bdd9]

goroutine 598 [running]:
github.com/wealdtech/chaind/services/eth1deposits/getlogs.(*Service).depositFromLogEntry(0xc000123810?, {0xe1e320, 0xc014bd4c60}, 0xc00768cbe0, 0x0, 0xc025273560)
        /home/henridf/chaind/services/eth1deposits/getlogs/handler.go:134 +0x199
github.com/wealdtech/chaind/services/eth1deposits/getlogs.(*Service).handleBlocks(0xc000123810, {0xe1e320, 0xc003febe00}, 0xb72300, 0xb7233f)
        /home/henridf/chaind/services/eth1deposits/getlogs/handler.go:54 +0x1d8
github.com/wealdtech/chaind/services/eth1deposits/getlogs.(*Service).parseNewBlocks(0xc000123810, {0xe1e278, 0xc00007fdc0}, 0xc0003107c0)
        /home/henridf/chaind/services/eth1deposits/getlogs/service.go:234 +0x5e5
github.com/wealdtech/chaind/services/eth1deposits/getlogs.(*Service).updateAfterRestart(0xc000123810, {0xe1e278?, 0xc00007fdc0}, 0xffffffffffffffff)
        /home/henridf/chaind/services/eth1deposits/getlogs/service.go:155 +0x105
created by github.com/wealdtech/chaind/services/eth1deposits/getlogs.New
        /home/henridf/chaind/services/eth1deposits/getlogs/service.go:134 +0x955
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@ dev:
   - ensure only a single beacon committees update can be run at any time
   - optimize caching of beacon committees during block processing (thanks to @ariskk)
   - optimize fetching of validator balances when calculating epoch summaries (thanks to @henridf)
+  - avoid crash when execution node does not return transaction for which it supplied a log entry
 
 0.6.10
   - avoid crash with uninitialised metrics

--- a/services/eth1deposits/getlogs/handler.go
+++ b/services/eth1deposits/getlogs/handler.go
@@ -97,15 +97,15 @@ func (s *Service) handleMissed(ctx context.Context, md *metadata) {
 			failed++
 			cancel()
 			continue
-		} else {
-			log.Trace().Msg("Updated block")
-			// Remove this from the list of missed blocks.
-			missedBlocks := make([]uint64, len(md.MissedBlocks)-1)
-			copy(missedBlocks[:failed], md.MissedBlocks[:failed])
-			copy(missedBlocks[failed:], md.MissedBlocks[i+1:])
-			md.MissedBlocks = missedBlocks
-			i--
 		}
+
+		log.Trace().Msg("Updated block")
+		// Remove this from the list of missed blocks.
+		missedBlocks := make([]uint64, len(md.MissedBlocks)-1)
+		copy(missedBlocks[:failed], md.MissedBlocks[:failed])
+		copy(missedBlocks[failed:], md.MissedBlocks[i+1:])
+		md.MissedBlocks = missedBlocks
+		i--
 
 		if err := s.setMetadata(ctx, md); err != nil {
 			log.Error().Err(err).Msg("Failed to set metadata")

--- a/services/eth1deposits/getlogs/handler.go
+++ b/services/eth1deposits/getlogs/handler.go
@@ -16,6 +16,7 @@ package getlogs
 import (
 	"context"
 	"encoding/binary"
+	"fmt"
 	"time"
 
 	"github.com/attestantio/go-eth2-client/spec/phase0"
@@ -44,6 +45,10 @@ func (s *Service) handleBlocks(ctx context.Context, startBlock uint64, endBlock 
 		if err != nil {
 			cancel()
 			return errors.Wrap(err, "failed to obtain transaction from transaction hash")
+		}
+		if tx == nil {
+			cancel()
+			return fmt.Errorf("no transaction returned for hash %#x", logEntry.TransactionHash)
 		}
 		receipt, err := s.transactionReceiptByHash(ctx, logEntry.TransactionHash)
 		if err != nil {

